frontends/common/resolveReferences/resolveReferences.h

```diff
@@ -32,46 +32,24 @@ enum class ResolutionType {
 };
 
 /// Data structure representing a stack of nested namespaces.
-class ResolutionContext : public IHasDbPrint {
-    /// Stack of nested namespaces
-    std::vector<const IR::INamespace*> stack;
-    /// Root namespace for the program.
-    const IR::INamespace* rootNamespace;
+class ResolutionContext : public IHasDbPrint, virtual public Visitor {
+ protected:
     /// Stack of namespaces for global declarations (e.g., match_kind)
     /// @todo: what about errors?
     std::vector<const IR::INamespace*> globals;
 
- public:
-    explicit ResolutionContext(const IR::INamespace* rootNamespace) :
-            rootNamespace(rootNamespace)
-    { push(rootNamespace); }
+    std::vector<const IR::IDeclaration*>*
```

Comment:
I think that the above todo is no longer necessary.

Reply:
I haven't yet tracked down what happens with errors -- it seems that we currently require them to be explicit (eg, `error.NoMatch` rather than just `NoMatch`).